pkg/apps/blackduck/latest/creater.go

```diff
@@ -71,90 +71,99 @@ func (hc *Creater) Ensure(blackduck *blackduckapi.Blackduck) error {
 
 	pvcs := hc.GetPVC(blackduck)
 
-	commonConfig := crdupdater.NewCRUDComponents(hc.KubeConfig, hc.KubeClient, hc.Config.DryRun, blackduck.Spec.Namespace,
-		&api.ComponentList{PersistentVolumeClaims: pvcs}, "app=blackduck,component=pvc")
-	errors := commonConfig.CRUDComponents()
-	if len(errors) > 0 {
-		return fmt.Errorf("update pvc: %+v", errors)
-	}
+	if strings.EqualFold(blackduck.Spec.DesiredState, "STOP") {
+		commonConfig := crdupdater.NewCRUDComponents(hc.KubeConfig, hc.KubeClient, hc.Config.DryRun, blackduck.Spec.Namespace,
+			&api.ComponentList{}, "app=blackduck")
+		errors := commonConfig.CRUDComponents()
+		if len(errors) > 0 {
+			return fmt.Errorf("update pvc: %+v", errors)
+		}
+	} else {
```

Comment:
instead of having the huge line of blocks in else statement, can we return the stop once it is done.

Reply:
No because there are other things (collect PV) that will need to be executed that are outside of the if else block.